build failure

When trying to use `update_deps.py` in a release build, I get the following error from trying to build spirv-tools:

```
/home/zmike/src/Vulkan-ValidationLayers/build/SPIRV-Tools/source/text_handler.cpp: In function ‘spv_result_t spvtools::{anonymous}::getWord(spv_text, spv_position, std::string*)’:
/home/zmike/src/Vulkan-ValidationLayers/build/SPIRV-Tools/source/text_handler.cpp:122:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
  122 |           if (escaping || quoting) break;
      |           ^~
/home/zmike/src/Vulkan-ValidationLayers/build/SPIRV-Tools/source/text_handler.cpp:124:9: note: here
  124 |         case '\0': {  // NOTE: End of word found!
      |         ^~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [source/CMakeFiles/SPIRV-Tools-shared.dir/build.make:432: source/CMakeFiles/SPIRV-Tools-shared.dir/text_handler.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 23%] Building CXX object source/CMakeFiles/SPIRV-Tools-static.dir/val/validate_atomics.cpp.o
[ 23%] Building CXX object source/CMakeFiles/SPIRV-Tools-static.dir/val/validate_barriers.cpp.o
/home/zmike/src/Vulkan-ValidationLayers/build/SPIRV-Tools/source/text_handler.cpp: In function ‘spv_result_t spvtools::{anonymous}::getWord(spv_text, spv_position, std::string*)’:
/home/zmike/src/Vulkan-ValidationLayers/build/SPIRV-Tools/source/text_handler.cpp:122:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
  122 |           if (escaping || quoting) break;
      |           ^~
/home/zmike/src/Vulkan-ValidationLayers/build/SPIRV-Tools/source/text_handler.cpp:124:9: note: here
  124 |         case '\0': {  // NOTE: End of word found!
      |         ^~~~
cc1plus: all warnings being treated as errors
```

I guess this means either not using -Werror for this subproject or resolving the warnings?

cmake 3.20.5
gcc 11.2.1

I tested your cmake command with a completely clean checkout and got this:

```
$ cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON -DUPDATE_DEPS=ON                           (~/src/Vulkan-ValidationLayers)
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib64/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.9.6", minimum required is "3") 
CMake Warning at CMakeLists.txt:58 (message):
  CMAKE_GENERATOR_PLATFORM not set.  Using x64 as target architecture.


********************************************************************************
* NOTE: Adding target vvl_update_deps to run as needed for updating            *
*       dependencies.                                                          *
********************************************************************************
CMake Error: Error: generator : Ninja
Does not match the generator used previously: Unix Makefiles
Either remove the CMakeCache.txt file and CMakeFiles directory or choose a different binary directory.
Starting builds in /home/zmike/src/Vulkan-ValidationLayers/external
Checking out glslang in /home/zmike/src/Vulkan-ValidationLayers/external/glslang
b'HEAD detached at 2fb89a00\nnothing to commit, working tree clean\n'
Building glslang in /home/zmike/src/Vulkan-ValidationLayers/external/glslang
Build dir = /home/zmike/src/Vulkan-ValidationLayers/external/glslang/build
Install dir = /home/zmike/src/Vulkan-ValidationLayers/external/glslang/build/install

CMake Error at CMakeLists.txt:95 (message):
  Could not run update_deps.py which is necessary to download dependencies.


-- Configuring incomplete, errors occurred!
See also "/home/zmike/src/Vulkan-ValidationLayers/build/CMakeFiles/CMakeOutput.log".
```

If I try to run `../scripts/update_deps.py --config release` from my build directory, I get the errors from the original post.

I use ninja as well, I guess this is just different because update_deps uses make?

Anyway, with `cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON -DUPDATE_DEPS=ON ` I now get the same exact error as in the OP:

```
FAILED: source/CMakeFiles/SPIRV-Tools-shared.dir/text_handler.cpp.o 
/usr/lib64/ccache/c++ -DSPIRV_CHECK_CONTEXT -DSPIRV_COLOR_TERMINAL -DSPIRV_LINUX -DSPIRV_TIMER_ENABLED -DSPIRV_TOOLS_IMPLEMENTATION -DSPIRV_TOOLS_SHAREDLIB -DSPIRV_Tools_shared_EXPORTS -I../ -I../include -I. -I../../SPIRV-Headers/include -O3 -DNDEBUG -fPIC -fvisibility=hidden -Wall -Wextra -Wnon-virtual-dtor -Wno-missing-field-initializers -Werror -std=c++11 -fno-exceptions -Wno-long-long -Wshadow -Wundef -Wconversion -Wno-sign-conversion -std=gnu++11 -MD -MT source/CMakeFiles/SPIRV-Tools-shared.dir/text_handler.cpp.o -MF source/CMakeFiles/SPIRV-Tools-shared.dir/text_handler.cpp.o.d -o source/CMakeFiles/SPIRV-Tools-shared.dir/text_handler.cpp.o -c ../source/text_handler.cpp
../source/text_handler.cpp: In function ‘spv_result_t spvtools::{anonymous}::getWord(spv_text, spv_position, std::string*)’:
../source/text_handler.cpp:122:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
  122 |           if (escaping || quoting) break;
      |           ^~
../source/text_handler.cpp:124:9: note: here
  124 |         case '\0': {  // NOTE: End of word found!
      |         ^~~~
cc1plus: all warnings being treated as errors
[46/319] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_atomics.cpp.o
[47/319] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_arithmetics.cpp.o
[48/319] Building CXX object source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_annotation.cpp.o
[49/319] Building CXX object source/CMakeFiles/SPIRV-Tools-static.dir/text_handler.cpp.o
FAILED: source/CMakeFiles/SPIRV-Tools-static.dir/text_handler.cpp.o 
/usr/lib64/ccache/c++ -DSPIRV_CHECK_CONTEXT -DSPIRV_COLOR_TERMINAL -DSPIRV_LINUX -DSPIRV_TIMER_ENABLED -I../ -I../include -I. -I../../SPIRV-Headers/include -O3 -DNDEBUG -fPIC -fno-rtti -Wall -Wextra -Wnon-virtual-dtor -Wno-missing-field-initializers -Werror -std=c++11 -fno-exceptions -Wno-long-long -Wshadow -Wundef -Wconversion -Wno-sign-conversion -std=gnu++11 -MD -MT source/CMakeFiles/SPIRV-Tools-static.dir/text_handler.cpp.o -MF source/CMakeFiles/SPIRV-Tools-static.dir/text_handler.cpp.o.d -o source/CMakeFiles/SPIRV-Tools-static.dir/text_handler.cpp.o -c ../source/text_handler.cpp
../source/text_handler.cpp: In function ‘spv_result_t spvtools::{anonymous}::getWord(spv_text, spv_position, std::string*)’:
../source/text_handler.cpp:122:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
  122 |           if (escaping || quoting) break;
      |           ^~
../source/text_handler.cpp:124:9: note: here
  124 |         case '\0': {  // NOTE: End of word found!
      |         ^~~~
cc1plus: all warnings being treated as errors
```

--- a/build-android/known_good.json
+++ b/build-android/known_good.json
@@ -28,7 +28,7 @@
       "name" : "SPIRV-Tools",
       "url" : "https://github.com/KhronosGroup/SPIRV-Tools.git",
       "sub_dir" : "shaderc/third_party/spirv-tools",
-      "commit" : "1fbed83c8aab8517d821fcb4164c08567951938f"
+      "commit" : "9529d3c2c647bf7c7aa1a2601d0209da98f8fa1e"
     },
     {
       "name" : "SPIRV-Headers",

--- a/layers/generated/spirv_tools_commit_id.h
+++ b/layers/generated/spirv_tools_commit_id.h
@@ -26,4 +26,4 @@
  ****************************************************************************/
 #pragma once
 
-#define SPIRV_TOOLS_COMMIT_ID "2fb89a0072ae7316af1c856f22663fde4928128a"
+#define SPIRV_TOOLS_COMMIT_ID "9529d3c2c647bf7c7aa1a2601d0209da98f8fa1e"

--- a/scripts/generate_source.py
+++ b/scripts/generate_source.py
@@ -79,7 +79,7 @@ def main(argv):
                  '-export_header'],
                 [common_codegen.repo_relative('scripts/external_revision_generator.py'),
                  '--json_file', common_codegen.repo_relative('scripts/known_good.json'),
-                 '--json_keys', 'repos,0,commit',
+                 '--json_keys', 'repos,3,commit',
                  '-s', 'SPIRV_TOOLS_COMMIT_ID',
                  '-o', 'spirv_tools_commit_id.h']]
 

--- a/scripts/known_good.json
+++ b/scripts/known_good.json
@@ -37,7 +37,7 @@
       "cmake_options": [
         "-DSPIRV-Headers_SOURCE_DIR={repo_dir}/../SPIRV-Headers"
       ],
-      "commit": "1fbed83c8aab8517d821fcb4164c08567951938f"
+      "commit": "9529d3c2c647bf7c7aa1a2601d0209da98f8fa1e"
     },
     {
       "name": "robin-hood-hashing",

